base/cvd/cuttlefish/host/commands/cvd/cli/commands/cache.cpp

```diff
@@ -0,0 +1,207 @@
+/*
+ * Copyright (C) 2024 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "host/commands/cvd/cli/commands/cache.h"
+
+#include <algorithm>
+#include <cstddef>
+#include <memory>
+#include <string>
+#include <string_view>
+#include <unordered_map>
+#include <vector>
+
+#include <android-base/logging.h>
+#include <fmt/format.h>
+
+#include "common/libs/utils/files.h"
+#include "common/libs/utils/flag_parser.h"
+#include "common/libs/utils/result.h"
+#include "common/libs/utils/tee_logging.h"
+#include "host/commands/cvd/cli/commands/command_handler.h"
+#include "host/commands/cvd/cli/types.h"
+#include "host/commands/cvd/utils/common.h"
+
+namespace cuttlefish {
+
+namespace {
+constexpr int kDefaultCacheSizeGigabytes = 25;
+
+constexpr char kDetailedHelpText[] =
+    R"(usage: cvd cache <action> [<flag>...]
+
+Example usage:
+    cvd cache where - display the filepath of the cache
+    cvd cache size - display the approximate size of the cache
```

Comment:
nit: These two can be merged into one, either just `cvd cache` or something like `cvd cache info`. It can format its output in a way that's easy to extract the information in a script:
```
$ cvd cache info
path: /tmp/cvd/123456/cache
size: 320G
```
so you can run `cvd cache info | grep ^path: | cut -d\  -f2` to obtain the path.
```